state/application_test.go

```diff
@@ -922,6 +922,16 @@ func (s *ApplicationSuite) TestUpdateConfigSettings(c *gc.C) {
 	}
 }
 
+func (s *ApplicationSuite) TestUpdateApplicationSeries(c *gc.C) {
```

Comment:
We need tests that check the behaviour when things change concurrently (eg adding subordinates). This is done using the various txn hook helpers See various uses of state.SetBeforeHooks, state.SetAfterHooks, state.SetRetryHooks